### PR TITLE
docs(core): expand e2e split guide

### PR DESCRIPTION
**Pull request by isaacplmann**
Expand [e2e task split guide](https://nx-dev-git-fork-isaacplmann-docs-e2e-splitting-nrwl.vercel.app/ci/features/split-e2e-tasks)
Expand [flaky task guide](https://nx-dev-git-fork-isaacplmann-docs-e2e-splitting-nrwl.vercel.app/ci/features/flaky-tasks)

### CODE REVIEW

File: docs/generated/packages/cypress/documents/overview.md [hunk 1]
```diff
@@ -75,6 +75,28 @@ The `@nx/cypress/plugin` is configured in the `plugins` array in `nx.json`.
...
+export default defineConfig({
+  e2e: {
+    ...nxE2EPreset(__filename, {
+      cypressDir: 'src',
+      bundler: 'vite',
+      webServerCommands: {
+        default: 'nx run my-project:serve',
+        production: 'nx run my-project:preview',
+      },
+      ciWebServerCommand: 'nx run my-project:serve-static',
```
Comment (barbados-clemens) [inline docs/generated/packages/cypress/documents/overview.md:93]: do people have to set it in the `nxE2EPreset` or not?

if they do then make sure to call that out, as some people don't use the preset.

File: docs/nx-cloud/features/flaky-tasks.md [hunk 2]
```diff
@@ -8,14 +8,18 @@ Nx is perfectly positioned to detect which tasks are flaky and automatically re-
...
 
-If you need to manually mark a task as flaky or not flaky, you can do so from the run details screen. Flaky tasks will have a button that says `Mark task as no longer flaky` and failed tasks that are not flaky will have a button that says `Mark task as likely flaky`. Using these buttons, you can ensure that Nx Cloud treats tasks in the appropriate way.
+## Manually Mark a Task as Flaky or Not Flaky
 
-![Mark task as no longer flaky button](/nx-cloud/features/mark-task-as-no-longer-flaky.png)
+If you don't want to wait for Nx Cloud to identify a task as flaky, you can manually mark a task as flaky from the run details screen. Failed tasks that are not flaky will have a button that says `Mark task as likely flaky`.
 
 ![Mark task as likely flaky button](/nx-cloud/features/mark-task-as-likely-flaky.png)
 
+Once you've resolved whatever issue caused a task to be flaky, you can manually mark a task as not flaky on the same run details screen. Flaky tasks will have a button that says `Mark task as no longer flaky`.
```
Comment (barbados-clemens) [inline docs/nx-cloud/features/flaky-tasks.md:19]:
```suggestion
Once you've resolved the issue caused a task to be flaky, you can manually mark a task as not flaky on the same run details screen. Flaky tasks will have a button that says `Mark task as no longer flaky`.
```

whatever feels slightly too casual for this.

File: docs/nx-cloud/features/split-e2e-tasks.md [hunk 3]
```diff
@@ -33,15 +33,301 @@ If you are already using the `@nx/cypress` or `@nx/playwright` plugin, you need
...
+  }
+}
+```
+
+{% /project-details %}
+
 You'll see that there are tasks named `e2e`, `e2e-ci` and a task for each e2e test file.
 
-Developers can run all e2e tests locally the same way as usual:
+Developers can run all e2e tests locally in the standard way:
```
Comment (barbados-clemens) [inline docs/nx-cloud/features/split-e2e-tasks.md:330]: should we say what the 'standard way' is here? personally, I think it assumes too much.

File: docs/nx-cloud/features/split-e2e-tasks.md [hunk 4]
```diff
@@ -55,8 +341,16 @@ nx e2e-ci my-project-e2e
...
+
+### More Precise Flaky Task Identification
 
-- Nx's cache can be used for all the e2e tasks that succeeded and only the failed tasks need to be re-run
-- Distributed Task Execution allows your e2e tests to be run on multiple machines simultaneously, which reduces the total time of the CI pipeline
-- Nx Agents can [automatically re-run failed flaky e2e tests](/ci/features/flaky-tasks) on a separate agent without a developer needing to manually re-run the CI pipeline
+Nx Agents can [automatically re-run failed flaky e2e tests](/ci/features/flaky-tasks) on a separate agent without a developer needing to manually re-run the CI pipeline. Without e2e task splitting, Nx can identify that the e2e tests in general are flaky. With e2e task splitting, Nx can identify which specific file is flaky - this way you can more quickly solve the problem.
```
Comment (barbados-clemens) [inline docs/nx-cloud/features/split-e2e-tasks.md:356]: this section reads a bit wonky to me. 

I think the `in general` is a bit off. 

and `can identify` should be `identifies` 

maybe?
```suggestion
Nx Agents [automatically re-run failed flaky e2e tests](/ci/features/flaky-tasks) on a separate agent without a developer needing to manually re-run the CI pipeline. 
Leveraging e2e task splitting, Nx identifies the specific flaky test file - this way you can quickly fix the offending test file. Without e2e splitting, Nx identifies only the e2e tests are flaky - requiring you to find the flaky test on your own.
```

idk if it's better just something to go off.

File: docs/nx-cloud/features/flaky-tasks.md [hunk 5]
```diff
@@ -8,14 +8,18 @@ Nx is perfectly positioned to detect which tasks are flaky and automatically re-
...
+
+![Mark task as no longer flaky button](/nx-cloud/features/mark-task-as-no-longer-flaky.png)
```
Comment (meeroslav) [inline docs/nx-cloud/features/flaky-tasks.md:21]: Why is this part of this PR?

File: docs/nx-cloud/features/flaky-tasks.md [hunk 6]
```diff
@@ -8,14 +8,18 @@ Nx is perfectly positioned to detect which tasks are flaky and automatically re-
...
 
-## Manually Mark a Task as Flaky or Not Flaky
+![Flaky tasks in CI](/nx-cloud/features/flaky-tasks-ci.png)
 
-If you need to manually mark a task as flaky or not flaky, you can do so from the run details screen. Flaky tasks will have a button that says `Mark task as no longer flaky` and failed tasks that are not flaky will have a button that says `Mark task as likely flaky`. Using these buttons, you can ensure that Nx Cloud treats tasks in the appropriate way.
+## Manually Mark a Task as Flaky or Not Flaky
 
-![Mark task as no longer flaky button](/nx-cloud/features/mark-task-as-no-longer-flaky.png)
+If you don't want to wait for Nx Cloud to identify a task as flaky, you can manually mark a task as flaky from the run details screen. Failed tasks that are not flaky will have a button that says `Mark task as likely flaky`.
```
Comment (meeroslav) [inline docs/nx-cloud/features/flaky-tasks.md:15]: Let's rephrase this to something like: "Once the flaky test is fixed, you can immediately mark it as no longer flaky by clicking on `Mark task as no longer flaky`"

Comment (meeroslav) [reply docs/nx-cloud/features/flaky-tasks.md:15]: But same as underneath, I would move those changes to a separate PR.

File: docs/nx-cloud/features/split-e2e-tasks.md (hunk 3 above)
Comment (meeroslav) [reply docs/nx-cloud/features/split-e2e-tasks.md:330]: Perhaps `non-distributed` way is more explicit?

**Review by meeroslav — CHANGES_REQUESTED** [review]
The `flaky-tasks-ci.png` shows test splitting, not flaky tasks so it should be named accordingly.

Additionally, you can clearly see that the second attempt succeeded so it's not really a good candidate to mark it as `no longer flaky`. We can crop the top for the `Flaky test` page where you don't see the body of the run.

Some other notes are inlined in comments.

**Comment by isaacplmann** [comment]
> The `flaky-tasks-ci.png` shows test splitting, not flaky tasks so it should be named accordingly.
> 
> Additionally, you can clearly see that the second attempt succeeded so it's not really a good candidate to mark it as `no longer flaky`. We can crop the top for the `Flaky test` page where you don't see the body of the run.
> 
> Some other notes are inlined in comments.

The `flaky-tasks-ci.png` image shows both test splitting and flaky tests.  The image shows a flaky task that is retried and succeeds on the second attempt.  It also shows the two tabs with logs for each attempt.  I added in text to explain that.

**Review by meeroslav — APPROVED** [review]
Approved to unblock it, but still expect those changes.
